src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleArray.cs

```diff
@@ -149,83 +150,110 @@ public static partial class JsonSerializer
         }
 
         // If this method is changed, also change ApplyValueToEnumerable.
-        internal static void ApplyObjectToEnumerable(object value, JsonSerializerOptions options, ref ReadStackFrame frame, bool setPropertyDirectly = false)
+        internal static void ApplyObjectToEnumerable(
+            object value,
+            JsonSerializerOptions options,
+            ref ReadStack state,
```

Comment:
Why make this change? It looks like there is no other usage of `state` other than `state.Current` here.

If you think this is cleaner, should we cache `state.Current` in a local?

Reply:
It's due to throwing an exception. It requires `state.PropertyPath`.

state.Current can't be in a local because it is a struct and a copy will be made.

Reply:
Oh... `Current` is a field on the struct. Can it be a auto-property?

I don't understand why it has to be a field:
https://github.com/dotnet/corefx/blob/07fdefad609cb4fb374366d12f69ce4011ef0329/src/System.Text.Json/src/System/Text/Json/Serialization/ReadStack.cs#L12-L13

Reply:
Because properties have value semantics...

```cs
    public struct MyStruct
    {
        public MyInnerStruct Prop { get; set; }
        public MyInnerStruct Field;
    }

    public struct MyInnerStruct
    {
        public int X;
    }

    class Program
    {
        static void Main(string[] args)
        {
            var s = new MyStruct();

            var inner = s.Prop;
            inner.X++;

            s.Field.X++;

            Console.WriteLine($"Prop: {s.Prop.X} Field: {s.Field.X}");
            // Output: Prop: 0 Field: 1

            Console.ReadLine();
        }
    }
```

Reply:
and `s.Prop.X++` doesn't compile for that reason.

Reply:
I see. Thanks for the explanation.